src/main/scala/io/iohk/ethereum/network/PeerEventBusActor.scala

```diff
@@ -0,0 +1,131 @@
+package io.iohk.ethereum.network
+
+import akka.actor.{Actor, ActorRef, Props}
+import akka.event.ActorEventBus
+import io.iohk.ethereum.network.PeerEventBusActor.PeerEvent.{MessageFromPeer, MessageToPeer, PeerDisconnected, PeerHandshakeSuccessful}
+import io.iohk.ethereum.network.PeerEventBusActor.SubscriptionClassifier._
+import io.iohk.ethereum.network.handshaker.Handshaker.HandshakeResult
+import io.iohk.ethereum.network.p2p.Message
+
+object PeerEventBusActor {
+  def props: Props = Props(new PeerEventBusActor)
+
+  sealed trait PeerSelector {
+    def contains(peerId: PeerId): Boolean
+  }
+
+  object PeerSelector {
+
+    case object AllPeers extends PeerSelector {
+      override def contains(p: PeerId): Boolean = true
+    }
+
+    case class WithId(peerId: PeerId) extends PeerSelector {
+      override def contains(p: PeerId): Boolean = p == peerId
+    }
+
+  }
+
+  sealed trait SubscriptionClassifier
+
+  object SubscriptionClassifier {
+    case class MessageReceivedClassifier(messageCodes: Set[Int], peerSelector: PeerSelector) extends SubscriptionClassifier
+    case class MessageSentClassifier(messageCodes: Set[Int], peerSelector: PeerSelector) extends SubscriptionClassifier
```

Comment:
Right, but we use explicit coupling of actor refs for events we generate everywhere else. We have to treat the incoming messages as events. Now if there's strong case against it, I'm all ears. but PeersInfoHolderActor can be the application level PeerManager it doesn't need to be passive although that would have been nice.